### ISSUE TITLE
Make it possible to use `NavigationViewTrackingStrategy` for dynamic `NavHostFragment`

### ISSUE THREAD

**Issue by tomaz** [DataDog/dd-sdk-android#760]
**Are you requesting automatic instrumentation for a framework or library? Please describe.**
/

**Is your feature request related to a problem? Please describe.**
`NavigationViewTrackingStrategy` only attempts to find nav controller in `onActivityStarted`, which is called before dynamically created and inserted `NavHostFragment.onCreate` (in our app, `onCreate` is called from within parent activity `onStart`, after activity lifecycle `onActivityStarted`). Since `NavController` is created in `Fragment.onCreate`, it's still null at this point. Because `NavigationViewTrackingStrategy` doesn't allow any other means of injecting `NavController`, we can't use it.

**Describe the solution you'd like**
Ideally, `NavigationViewTrackingStrategy` should have alternate method of injecting `NavController`. For example as optional public property or a setup function:

```
class NavigationViewTrackingStrategy(
    ...
) ... {

    private var navController: NavController? = null
    ...

    // region ActivityLifecycleTrackingStrategy

    override fun onActivityStarted(activity: Activity) {
        super.onActivityStarted(activity)
        setup(activity.findNavControllerOrNull(navigationViewId))
    }

    ...

    fun setup(controller: NavController?) {
        navController?.removeOnDestinationChangedListener(this)

        navController = controller

        navController?.let {
            if (FragmentActivity::class.java.isAssignableFrom(activity::class.java)) {
                val navControllerFragmentCallbacks = NavControllerFragmentLifecycleCallbacks(
                    it,
                    argumentsProvider = { emptyMap() },
                    componentPredicate = predicate
                )
                navControllerFragmentCallbacks.register(activity as FragmentActivity)
                lifecycleCallbackRefs[activity] = navControllerFragmentCallbacks
            }
            it.addOnDestinationChangedListener(this)
        }
    }
```

**Describe alternatives you've considered**
We're implementing our own tracking strategy, which is a shame since `NavigationViewTrackingStrategy` provides so much out of the box...

**Additional context**
/

**Comment by xgouchet** [comment]
Hi @tomaz, that's indeed a use case we haven't considered yet. Since you're going to write your own Strategy, feel free to submit it as an `LazyNavigationViewTrackingStrategy` in a PR. That we we can enhance it whenever we add new features to the SDK, and it could be used by other customers as well.

**Comment by tomaz** [comment]
Hey! I will likely not be able to submit the concrete implementation due to using private code, but will look into PR with more generic solution afterwards. First I need to get logs working though, which for some reason is more challenging that anticipated 😅

**Comment by hon92** [comment]
Hi.

today i found that issue in 1.18.1 and if i understand correctly, issue is due to using new FragmentContainerView for our NavHostFragment which we should use according to google documentation. 

So this code is not working properly when app is start (which will cause only ApplicationLaunch view in DD RUM session)

```@Suppress("SwallowedException")
    private fun Activity.findNavControllerOrNull(@IdRes viewId: Int): NavController? {
        return try {
            findNavController(viewId) // TODO: this is no longer valid code
        } catch (e: IllegalArgumentException) {
            null
        } catch (e: IllegalStateException) {
            null
        }
    }
```
I suggest just quick fix for that. 

Use `(activity as (AppCompatActivity)).supportFragmentManager.findFragmentById(viewId)?.findNavController()` instead of `findNavController(viewId)` in that function. 

Please take a look if that is correct solution and proceed with PR. 
Currently because of this issue is NavigationViewTrackingStrategy quite useless due to this issue which should be fixed long time ago. Please fix this urgently (i dont wanna write custom view tracking strategy :) )

**Comment by 0xnm** [comment]
Hello @hon92!

Can you please elaborate the following statement:

> issue is due to using new FragmentContainerView for our NavHostFragment which we should use according to google documentation

If you could give some examples of your setup/links to the documentation you mentioned, it would be great.

**Comment by plousada** [comment]
Hi @hon92 , thank you for providing more information and for the explanation. 

It looks like this is a good approach. We'll do some internal testing for backwards compatibility and some edge cases and we'll try to release a fix for this soon.

**Comment by 0xnm** [comment]
Hello @hon92! I did the necessary change in https://github.com/DataDog/dd-sdk-android/pull/1538, it is available in version `2.0.0-beta2`, you can check release details [here](https://github.com/DataDog/dd-sdk-android/releases/tag/2.0.0-beta2).

@tomaz I don't know your setup, but if you used `FragmentContainerView` instead of just `fragment` for hosting navigation controller, then maybe this change will work for you as well, removing the need to call `startTracking` manually.
